rp235x-hal-examples/README.md

```diff
@@ -56,43 +56,156 @@ https://github.com/rp-rs/rp-hal-boards/ for more details.
-   Compiling rp235x-hal-examples v0.1.0 (/home/user/rp-hal/rp235x-hal-examples)
-    Finished `dev` profile [unoptimized + debuginfo] target(s) in 4.53s
 $ cargo build --bin blinky
-    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
+   Compiling proc-macro2 v1.0.89
+   Compiling unicode-ident v1.0.13
+   Compiling syn v1.0.109
+...
+   Compiling pio-parser v0.2.2
+   Compiling rp235x-hal v0.2.0 (/home/wink/prgs/rpi-pico/myrepos/rp-hal-clone/rp235x-hal)
+   Compiling pio-proc v0.2.2
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 14.97s
+```
+
+This builds the default target, which is the for the ARM at `./target/thumbv8m.main-none-eabihf/debug/blinky`:
+```console
 $ file ./target/thumbv8m.main-none-eabihf/debug/blinky
 ./target/thumbv8m.main-none-eabihf/debug/blinky: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, with debug_info, not stripped
 ```
 
-You can also 'run' an example, which thanks to our supplied
-[`.cargo/config.toml`](./.cargo/config.toml) will invoke Raspberry Pi's
-`picotool` to copy it to an RP235x in USB Bootloader mode. You should install
-that if you don't have it already, from
-<https://github.com/raspberrypi/pico-sdk-tools/releases>.
+If you want to build the RISC-V you can specify the target directly to override the default:
+```console
+$ cargo build --target=riscv32imac-unknown-none-elf --bin blinky
+   Compiling nb v1.1.0
+   Compiling byteorder v1.5.0
+   Compiling stable_deref_trait v1.2.0
+..
+   Compiling futures v0.3.31
+   Compiling frunk v0.4.3
+   Compiling rp235x-hal v0.2.0 (/home/wink/prgs/rpi-pico/myrepos/rp-hal-clone/rp235x-hal)
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 6.23s```
+```
+
+And we see the RISC-V at `./target/riscv32imac-unknown-none-elf/debug/blinky`:
+```console
+$ file ./target/riscv32imac-unknown-none-elf/debug/blinky
+./target/riscv32imac-unknown-none-elf/debug/blinky: ELF 32-bit LSB executable, UCB RISC-V, RVC, soft-float ABI, version 1 (SYSV), statically linked, with debug_info, not stripped
+```
+
+You can also specify the ARM target directly just pass
+`--target thumbv8m.main-none-eabihf` instead of `--target riscv32imac-unknown-none-elf`.
+
+You can also easily build, flash and start the application on the RP235x
+by using the `cargo run` using command:
+```console
+$ cargo run --target thumbv8m.main-none-eabihf --bin blinky
+$ cargo run --target riscv32imac-unknown-none-elf --bin blinky
+```
+
+You can also specify a release build by passing `--release` to the `cargo build`
+or `cargo run` commands. This will build the example with optimizations enabled:
+```console
+$ cargo run --target thumbv8m.main-none-eabihf --release --bin blinky
+$ cargo run --target riscv32imac-unknown-none-elf --release --bin blinky
+```
+
+The above work well but the commands are somewhat verbose. To make building and running
+more succinct we have added some aliases in .carog/config.toml:
+
+```toml
```

Comment:
Link to the file rather than copy-pasting the contents.

Reply:
OK